### PR TITLE
Add Swift Package Manager support

### PR DESCRIPTION
**Pull request by zocario**
Hello I'm trying to give a little help regarding support of Swift Package Manager as I need it in one of my project :) 
This is a very simple implementation I just did and allowed me to use Family with SPM recently. I'm not sure if this is good enough but I think it is a good base to start with if it helps.

### Changes
* Create target for shared code between macOS and mobile platforms (iOS & tvOS) named Family-Shared. 
This target is used by both macOS and mobile target. 
* Family-mobile encapsulate both tvOS and iOS frameworks.

### CODE REVIEW

**Comment by zenangst** [comment]
@zocario  Good start mate and thanks for starting to add support for this.
Seems like there is still some additional work to be done on the macOS target, mind taking a look at that?

**Comment by zocario** [comment]
@zenangst Not today but I can have a look for macOS at the end of week :)

**Comment by zenangst** [comment]
No worries, I just check to see why Travis wasn't playing ball. Cheers!

**Comment by zocario** [comment]
I also forgot to add tests targets, I'll polish this and mention you again ;)

**Comment by zocario** [comment]
@zenangst I had a second look and fixed macOS compilation when opening swift package. 
But I'm facing two failures test on iOS+tvOS: 
* testAddingMultipleChildViewControllers
* testViewControllerIsVisibleMethods

I'm not really sure to see how this has changed with my modifications to be honest. If you have time to give your opinion on the changes I've started don't hesitate :)

**Comment by zocario** [comment]
Also to me we shouldn't need to use the xcodeproj anymore, but I'm not sure we can work without it regarding travis CI. I'll need to update the xcodeproj to work with the changes I made I'll update this soon.

**Comment by zocario** [comment]
@zenangst So I succeed to setup Travis by using the demo project as container of the swift package. It is using the local swift package and exposes its schemes to be run by Travis. I don't know if this setup would be fine for you (I think it should work for Carthage and Cocoapods but needs to confirm) but this is how I'm doing currently at work to get CI running with swift packages. The constraint is just that the xcodeproj can't be at the root level with this configuration that's why I've used the demo project.

**Comment by zenangst** [comment]
Looking good, I was actually thinking about removing the project all together and perhaps use `XcodeGen` instead so I think that change would be fine.

**Comment by zenangst** [comment]
Don't have time to do a full review now but perhaps early next week if again… life allows 😎

**Comment by zenangst** [comment]
@zocario mind fixing up the merge conflicts here? (totally my bad 😶)

**Comment by zocario** [comment]
@zenangst Done :)

**Comment by zocario** [comment]
Also I have a test failure on Mac OS but nothing fails locally for me (10.15.7 Travis is 10.15.6), would you mind trying on your side?

File: Tests/macOS/FamilyViewControllerSharedTests.swift [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        XCTAssertEqual(margins.left, insets.left)
+        XCTAssertEqual(margins.right, insets.right)
+        XCTAssertEqual(margins.bottom, insets.bottom)
+
+        XCTAssertEqual(padding.top, insets.top)
+        XCTAssertEqual(padding.left, insets.left)
+        XCTAssertEqual(padding.right, insets.right)
+        XCTAssertEqual(padding.bottom, insets.bottom)
+
+        XCTAssertEqual(container.scrollView.bounds, CGRect(origin: .zero, size: CGSize(width: 500, height: 1000)))
```
Comment (zenangst) [inline Tests/macOS/FamilyViewControllerSharedTests.swift:45]: I removed this line on `master`, if you do the same I think your branch should work.

**Comment by zenangst** [comment]
Hey @zocario, I tried this branch and I was unable to install it using CocoaPods.
I ended up with the following error inside of `FamilyScrollView`

image

**Comment by zenangst** [comment]
I'm curious if we could refactor the `Package.swift` to only produce one framework per platform. I think that would fix the issue with the module issue that arises when trying to use it with CocoaPods.

**Comment by zocario** [comment]
Hey @zenangst I'm sorry I completely missed your last message, I'll take an other look this week-end, big weeks at work at the moment 🙈

**Comment by zenangst** [comment]
@zocario no worries mate :)

**Comment by zenangst** [comment]
Hmm, I'm still having issues installing this using CocoaPods

**Comment by zenangst** [comment]
@zocario I opened a new PR here - https://github.com/zenangst/Family/pull/207 that takes a different approach. Instead of creating a shared framework, it uses pre-conditions per platform. The major benefit here is that it still works when using it with CocoaPods.

**Comment by zocario** [comment]
@zenangst Looks great! Way better approach that my try :) To be honest I wasn't sure about how to do this in an other way, and I had a lots of things to do at work. I should have thought about pre-conditions! I close the PR ;)

**Comment by zenangst** [comment]
@zocario no worries mate, thanks for all the hard work that you put into this! ❤️

**Pull request closed without merging**